src/App.tsx

```diff
@@ -1,73 +1,99 @@
-import React from 'react';
+import debounce from 'lodash.debounce';
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+import { DropDown } from './components/DropDown';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [people, setPeople] = useState<Person[]>(peopleFromServer);
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+
+  const inputValue = useRef<HTMLInputElement | null>(null);
+  const applyQuery = useCallback(debounce(setAppliedQuery, 500), []);
```

Comment:
The `debounce` function is used here to delay the `setAppliedQuery` call, but it should be noted that `debounce` does not automatically pass arguments to the debounced function. You need to ensure that the `event.target.value` is correctly passed to `setAppliedQuery` when `applyQuery` is called.